packages/optimistic-oracle/src/mappings/optimisticOracle.ts

```diff
@@ -225,3 +243,23 @@ export function handleOptimisticSettle(event: Settle): void {
 
   request.save();
 }
+
+export function handleSetBond(call: SetBondCall): void {
+  log.warning(`OO set bond inputs: {},{},{},{}`, [
+    call.inputs.timestamp.toString(),
+    call.inputs.identifier.toString(),
+    call.inputs.ancillaryData.toHex(),
+    call.inputs.bond.toString(),
+  ]);
+  let requestId = call.inputs.identifier
+    .toString()
+    .concat("-")
+    .concat(call.inputs.timestamp.toString())
+    .concat("-")
+    .concat(call.inputs.ancillaryData.toHex());
+
+  let request = getOrCreateOptimisticPriceRequest(requestId);
+  request.finalFee = call.outputs.totalBond;
```

Comment:
this function is confusing, called setBond but then setting finalFee, whats the deal with that?
also what is totalBond? is this calculated the same way as this `requestData.finalFee.plus(requestData.bond);`

Reply:
Hey @daywiss, I understand the confusion. In OOv1, we didn't define a bond and only used the finalFee. To maintain consistency and avoid frontend issues, I kept it that way. Upon examining the [parsePriceRequestGraphEntity](https://github.com/UMAprotocol/optimistic-oracle-dapp-v2/blob/28f141b77b572825363986b1f7735412d82c141a/shared/utils/graphql.ts#L105) function in oracle-dapp-v2, it appears that for v1 requests, we display the finalFee and set the bond to 0. Hence, I combined bond + final fee under finalFee. In this handler, totalBond is the sum of finalFee and bond.

Should we consider refactoring to include the bond? It would necessitate a change in the dapp. What are your thoughts?